Tasks/ExtractFiles/extractFiles.ts

```diff
@@ -0,0 +1,307 @@
+        result = repoRoot + path.sep + normalizedPath;
+        console.log('Relative file path: '+ normalizedPath+' resolving to: ' + result);
+    }
+    return result;
+}
+
+// Find matching archive files
+var files: string [] = findFiles();
+console.log('Found: ' + files.length + ' files to extract:');
+for (var i = 0; i < files.length; i++) {
+    console.log(files[i]);
+}
+
+// Clean the destination folder before extraction?
+if (cleanDestinationFolder && tl.exist(destinationFolder)) {
+    console.log('Cleaning destination folder before extraction: ' + destinationFolder);
+    tl.rmRF(destinationFolder, false);
+}
+    
+// Create the destination folder if it doesn't exist
+if (!tl.exist(destinationFolder)) {
+    console.log('Creating destination folder: ' + destinationFolder);
+    tl.mkdirP(destinationFolder);
+}
+
+function isZip(file) {
+    return file.endsWith('.zip')
+        || file.endsWith('.jar')
+        || file.endsWith('.war')
+        || file.endsWith('.ear');
+}
+
+function isTar(file) {
+    var name = win ? file.toLowerCase() : file;
+    // standard gnu-tar extension formats with recognized auto compression formats
+    // https://www.gnu.org/software/tar/manual/html_section/tar_69.html
+    return name.endsWith('.tar')      // no compression
+        || name.endsWith('.tar.gz')   // gzip
+        || name.endsWith('.tgz')      // gzip
+        || name.endsWith('.taz')      // gzip
+        || name.endsWith('.tar.Z')    // compress
+        || (win && name.endsWith('tar.z')) // no case comparison for win
+        || name.endsWith('.taZ')      // compress // no case for win already handled above
+        || name.endsWith('.tar.bz2')  // bzip2
+        || name.endsWith('.tz2')      // bzip2
+        || name.endsWith('.tbz2')     // bzip2
+        || name.endsWith('.tbz')      // bzip2
+        || name.endsWith('.tar.lz')   // lzip
+        || name.endsWith('.tar.lzma') // lzma
+        || name.endsWith('.tlz')      // lzma
+        || name.endsWith('.tar.lzo')  // lzop
+        || name.endsWith('.tar.xz')   // xz
+        || name.endsWith('.txz');     // xz
+}
+
+function unzipExtract(file, destinationFolder) {
+    console.log('Extracting file: ' + file);
+    if(typeof xpUnzipLocation == "undefined") {
+        xpUnzipLocation = tl.which('unzip', true);
+    }
+    var unzip = tl.createToolRunner(xpUnzipLocation);
+    unzip.arg(file);
+    unzip.arg('-d');
+    unzip.arg(destinationFolder);
+    return handleExecResult(unzip.execSync(), file);
+}
+
+function sevenZipExtract(file, destinationFolder) {
+    console.log('Extracting file: ' + file);
+    var sevenZip = tl.createToolRunner(getSevenZipLocation());
+    sevenZip.arg('x');
+    sevenZip.arg('-o' + destinationFolder);
+    sevenZip.arg(file);
+    return handleExecResult(sevenZip.execSync(), file);
+}
+
+function tarExtract(file, destinationFolder) {
+    console.log('Extracting file: ' + file);
+    if (typeof xpTarLocation == "undefined") {
```

Comment:
Instead of the comparison to `undefined`, you should also be able to use `if (!xpTarLocation)`.